src/solvers/refinement/string_constraint_generator_constants.cpp

```diff
@@ -67,6 +67,39 @@ exprt string_constraint_generatort::add_axioms_for_empty_string(
   return from_integer(0, get_return_code_type());
 }
 
+/// Convert an expression of type string_typet to a string_exprt
+/// \param res: string expression for the result
+/// \param arg: expression of type string typet
+/// \param guard: condition under which `res` should be equal to arg
+/// \return 0 if constraints were added, 1 if expression could not be handled
+///         and no constraint was added. Expression we can handle are of the
+///         form \f$ e := "<string constant>" | (expr)? e : e \f$
+exprt string_constraint_generatort::add_axioms_for_cprover_string(
+  const array_string_exprt &res,
+  const exprt &arg,
+  const exprt &guard)
+{
+  if(const auto if_expr = expr_try_dynamic_cast<if_exprt>(arg))
+  {
+    const and_exprt guard_true(guard, if_expr->cond());
+    const exprt return_code_true =
+      add_axioms_for_cprover_string(res, if_expr->true_case(), guard_true);
+
+    const and_exprt guard_false(guard, not_exprt(if_expr->cond()));
+    const exprt return_code_false =
+      add_axioms_for_cprover_string(res, if_expr->false_case(), guard_false);
+
+    return if_exprt(
+      equal_exprt(return_code_true, from_integer(0, get_return_code_type())),
+      return_code_false,
+      return_code_true);
+  }
+  else if(const auto constant_expr = expr_try_dynamic_cast<constant_exprt>(arg))
+    return add_axioms_for_constant(res, constant_expr->get_value(), guard);
+  else
```

Comment:
A comment explaining why in this case a `1` expression is a reasonable return might be good.